lib/omniauth/strategies/open_id.rb

```diff
@@ -59,13 +60,21 @@ def start
         end
       end
 
-      def get_identifier
+      def openid_form
         f = OmniAuth::Form.new(:title => 'OpenID Authentication')
         f.label_field('OpenID Identifier', options.identifier_param)
         f.input_field('url', options.identifier_param)
         f.to_response
       end
 
+      def get_identifier
+        if options[:login_entrypoint].respond_to?(:call)
+          instance_eval &options[:login_entrypoint]
```

Comment:
Why not `options[:login_entrypoint].call(env)`? That would follow the existing art in omniauth-identity - https://github.com/intridea/omniauth-identity/blob/master/lib/omniauth/strategies/identity.rb#L15-L28